src/newsView.js

```diff
@@ -0,0 +1,91 @@
+const NewsApi = require("./newsApi")
+const NewsModel = require("./newsModel")
+
+class NewsView{
+	
+	constructor(model = new NewsModel(), api = new NewsApi()){
+		this.model = model;
+		this.api = api;
+		this.newsBlock = document.querySelector('.news-block');
+		
+
+		(()=>{
+			this.load(()=>{
+				let btn = document.querySelectorAll('.my-btn');
+				this.newsSlot = document.querySelector('#news-slot');
+				window.addEventListener('hashchange', (e) => this.routechange(e))
+				btn.forEach(el =>{
+					el.addEventListener('click', ()=>{
+						// this.newsSlot.style.display = 'flex'
+						this.newsSlot.style.height = '320px'
+						this.newsSlot.style.opacity = 1
+						console.log('button clicked');
+					})
+				})
+			})
+		})();
+
+	}
+
+	routechange(e){
+		const hashLocation = window.location.hash.substring(1);
+		console.log(hashLocation);
+		this.loadPopUp(hashLocation)
+	}
+
+	loadPopUp(url){
+		this.api.fetchNews((data)=>{
+			let result = data[url - 1].webTitle;
```

Comment:
Being picky here but could we use a better name than result? Just so we know what data[url - 1].webTitle is meant to represent.